src/meltano/cli/state.py

```diff
@@ -309,13 +309,35 @@ def get_state(ctx: click.Context, project: Project, state_id: str) -> None:
 
 
 @meltano_state.command(cls=InstrumentedCmd, name="clear")
-@prompt_for_confirmation(prompt="This will clear state for the job. Continue?")
-@click.argument("state-id")
+@prompt_for_confirmation(prompt="This will clear state for the job(s). Continue?")
+@click.argument("state-id", required=False)
+@click.option(
+    "--all",
+    "clear_all",
+    is_flag=True,
+    required=False,
+    help="Clear all states IDs.",
+)
 @pass_project(migrate=True)
 @click.pass_context
-def clear_state(ctx: click.Context, project: Project, state_id: str) -> None:
+def clear_state(
+    ctx: click.Context,
+    project: Project,
+    state_id: str,
+    clear_all: bool,  # noqa: FBT001
+) -> None:
     """Clear state."""
-    state_service: StateService = (
-        state_service_from_state_id(project, state_id) or ctx.obj[STATE_SERVICE_KEY]
-    )
-    state_service.clear_state(state_id)
+    # Case where neither or both have been provided
+    if bool(state_id) == clear_all:  # pragma: no cover
```

Comment:
This seems a bit clearer

```suggestion
    if state_id and clear_all:  # pragma: no cover
```

Reply:
I've pushed 13e2e9eba1d434c75d5d0721dcd4044543bfee6e to actually test this code branch. The idea is that the sub-command should fail when the user

- provides both the `state-id` positional arg and the `--all` flag
- doesn't provide the `state-id` positional arg nor the `--all` flag

Does that make sense? This may be more readable?

```python
if (state_id is None and not clear_all) or (state_id is not None and clear_all):
```

Reply:
Ah right, this is an XNOR operation, and `a XNOR b` is equivalent to `bool(a) == bool(b)`. I'd say it's fine as-is.